SimpleVM/metadata-service-roles
# Feat/package

@dweinholz 

How to test:
- Start a new machine in staging environment (ubuntu 22.04 due to the current state of apt.bi.denbi.de)
- Activate public-key propagation for thie machine

on staging currently the metadata-endpoint is set incorrectly and the machine name is not allowed to contain any capital letters..., so please test with a machine without capital letters (fix is on dev already)

and please adjust in the metadata_config.env:
`METADATA_INFO_ENDPOINT=https://simplevm-dev.bi.denbi.de/portal/api/metadata-info/`


- then finally run the following commands to add the repository

```
wget https://apt.bi.denbi.de/repo_key.key
sudo apt-key  add repo_key.key
sudo add-apt-repository https://apt.bi.denbi.de/repos/apt/jammy jammy main
sudo apt update

sudo apt install simplevm-metadata-service
```

This will install the core package and the key-sync package.
You can then check the state of the service started:

`sudo systemctl status public-keys-sync.service`
Should show when the service got triggered last time.

You can also check `journalctl -xe` for errors, as well as the log-file at `/var/log/metadata.log`.

the file `/etc/simplevm/metadata_config.env` will then hold the compatibility json endpoint which is used by check_version.sh

Try to add and remove keys, the synchonization should take place every 2 minutes.
The keys will be found in `~/.ssh/metadata_authorized_keys`.
